docs/1-introduction/1-3-models/1-3-1-texturing.md

```diff
@@ -0,0 +1,162 @@
+- D3D11_USAGE_STAGING	: "This texture is copied to/from often (either on CPU or GPU)" (This 'only' supports copying, not arbitary writing).
+
+`BindFlags` will tell D3D11 how we're going to use the texture, there are multiple values that do not make sense (or are valid) for textures, so we'll only explain the valid ones:
+
+- D3D11_BIND_SHADER_RESOURCE	: This a read-only resource for shaders (This is the one we use for regular textures!)
+- D3D11_BIND_STREAM_OUTPUT		: This is for our swapchain output, very rarely used.
+- D3D11_BIND_RENDER_TARGET		: This a write-only resource (render target) 
+- D3D11_BIND_DEPTH_STENCIL		: This is a special (write-only) DepthStencil resource (use for depth buffers)
+- D3D11_BIND_UNORDERED_ACCESS	: This is a read-write resource for shaders.
+
+Because we don't have plans to change the texture after creation and we only use it as a read-only resource for shaders, we set Usage to `D3D11_USAGE_IMMUTABLE` and BindFlags to `D3D11_BIND_SHADER_RESOURCE`
+However because it is IMMUTABLE we need to supply the data immediately upon creation, we'll do this with a `D3D11_SUBRESOURCE_DATA` parameter.
+This is a simple structure that tells D3D11 where to find the data and how big it is.
+For `pSysMem` we point it to the data that FreeImage is holding which we can get with: `FreeImage_GetBits(image)`.
+`SysMemPitch` will tell D3D11 how big a row of data is, as we have only told it how to interpret the data in `D3D11_TEXTURE2D_DESC` but not what it 'actually' is.
+The size of a row of data from a 32-bit texture with a width of 512 is calculated as following: 
+We can get the amount of channels (we're assuming 8 bits each) by dividing the image's BPP (Bits Per Pixel) by 8 (so our resulting channel count is 4), we multiply this with the width, and so we end up with our example being:
+`(32/8) * 512 = 2048` bytes per row.
+
+And that's it.. now we can finally create our texture resource by calling: `device->CreateTexture2D(&textureDesc, &initialData, texture.GetAddressOf())`.
+Alas we're not there yet, we have created the **resource** but have not created our **view** which is what we need to bind to a **shader** stage.
+We do this by creating a **ShaderResourceView**, thankfully all the hard work is already done and we can simply fill in the few bits in a `D3D11_SHADER_RESOURCE_VIEW_DESC` by copying most of it from our `D3D11_TEXTURE2D_DESC`, the only thing new here is us telling it that the view should interpret the data as a 2D texture by specifying `D3D11_SRV_DIMENSION_TEXTURE2D`.
+
+We call `device->CreateShaderResourceView(texture.Get(), &srvDesc, &srv))` and end up with our most important piece, an `ID3D11ShaderResourceView`, at this point we no longer care about the original resource data (`WRL::ComPtr<ID3D11Texture2D> texture`) and let it go out of scope. This works because the ShaderResourceView hold an internal reference to the Texture2D so the data is not deleted until we get rid of our `ID3D11ShaderResourceView`.
```

Comment:
Usually you can get rid of the resource after creating the srv, do we do that here as well?

Reply:
`the ShaderResourceView hold` should be `the ShaderResourceView holds`